includes/html/graphs/application/sagan_global_bytes_stacked.inc.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+$name = 'sagan';
+$unit_text = 'Bytes';
+$colours = 'rainbow';
+$dostack = 0;
+$printtotal = 0;
+$nototal = 1;
+$addarea = 0;
+$transparency = 15;
+
+$rrd_list = [];
+
+foreach (\App\Models\Application::query()->where('app_type', 'sagan')->lazy() as $app) {
```

Comment:
To Avoid O(n) here, use this:
```php
$applications = \App\Models\Application::with('device')
    ->where('app_type', 'sagan')
    ->get();
```

and

`$app->device->hostname`